### PR TITLE
Functional Tests : Migration TS (BO > Advanced Parameters 08 => 10)

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Functional Tests : Migration TS (BO > Advanced Parameters 08 => 10)
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Relative to #30263
| Related PRs       | N/A
| How to test?      | Nightly is :green_circle:  : https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/3574407535
| Possible impacts? | N/A

### CODE REVIEW

File: tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts [hunk 1]
```diff
@@ -39,6 +29,11 @@ Sort logs table by : Id, Employee, Severity, Message, Object type, Object ID, Er
  */
 
 describe('BO - Advanced Parameters - Logs : Filter, sort and pagination logs table', async () => {
+  let browserContext: BrowserContext;
+  let page: Page;
+  let numberOfLogs: number = 0;
+  const today: string = getDateFormat('mm/dd/yyyy');
```
Comment (NeOMakinG) [inline tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts]: This type can be inferred by `getDateFormat` directly, which should be enough

File: tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts [hunk 2]
```diff
@@ -76,7 +71,7 @@ describe('BO - Advanced Parameters - Logs : Filter, sort and pagination logs tab
 
   // Login and logout 11 times to have 11 logs
   describe('Logout then login 11 times to have 11 logs', async () => {
-    const tests = new Array(11).fill(0, 0, 11);
+    const tests: number[] = new Array(11).fill(0, 0, 11);
```
Comment (NeOMakinG) [inline tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts:74]: Does it need to be typed too?

File: tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts [hunk 3]
```diff
@@ -93,14 +88,14 @@ describe('BO - Advanced Parameters - Logs : Filter, sort and pagination logs tab
 
       await dashboardPage.goToSubMenu(page, dashboardPage.advancedParametersLink, dashboardPage.logsLink);
 
-      const pageTitle = await logsPage.getPageTitle(page);
+      const pageTitle: string = await logsPage.getPageTitle(page);
```
Comment (NeOMakinG) [inline tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts]: as `logsPage.getPageTitle(page)` return should be typed as a string, this `string` type looks unnecessary

File: tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts [hunk 4]
```diff
@@ -93,14 +88,14 @@ describe('BO - Advanced Parameters - Logs : Filter, sort and pagination logs tab
...
       await expect(pageTitle).to.contains(logsPage.pageTitle);
     });
 
     it('should check the number of logs', async function () {
       await testContext.addContextItem(this, 'testIdentifier', 'checkLogsNumber', baseContext);
 
-      const numberOfElements = await logsPage.getNumberOfElementInGrid(page);
+      const numberOfElements: number = await logsPage.getNumberOfElementInGrid(page);
```
Comment (NeOMakinG) [inline tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts]: again, prefer adding the type to `logsPage.getNumberOfElementInGrid` return

**Review by NeOMakinG — COMMENTED** [review]
Added some comments that should be addressed everywhere inside the PR, most of the time you shouldn't need to type a variable as its type is normally inferred by the assignment

File: tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts (hunk 1 above)
Comment (Progi1984) [reply tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts]: Fixed

File: tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts (hunk 2 above)
Comment (Progi1984) [reply tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts:74]: Fixed

File: tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts (hunk 3 above)
Comment (Progi1984) [reply tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts]: Fixed

File: tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts (hunk 4 above)
Comment (Progi1984) [reply tests/UI/campaigns/functional/BO/14_advancedParameters/08_logs/01_filterSortAndPagination.ts]: Fixed

**Review by NeOMakinG — DISMISSED** [review]

**Review by nesrineabdmouleh — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by matks** [comment]
I am sorry @Progi1984 this PR needs a rebase 😢

**Review by matks — DISMISSED** [review]
Git conflicts, need rebase

**Comment by Progi1984** [comment]
@nesrineabdmouleh @matks @matthieu-rolland @NeOMakinG It's rebased

**Review by nesrineabdmouleh — DISMISSED** [review]

**Review by matks — APPROVED** [review]

**Review by matks — DISMISSED** [review]

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by SD1982 — APPROVED** [review]

**Comment by eternoendless** [comment]
Thanks @Progi1984

**Pull request merged**